src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AspNetCore6Plus/BrowserInjectingStreamWrapper.cs

```diff
@@ -39,7 +39,8 @@ public override Task FlushAsync(CancellationToken cancellationToken)
         {
             if (!Disabled && !_isContentLengthSet && IsHtmlResponse())
             {
-                _context.Response.ContentLength = null;
+                if (!_context.Response.HasStarted)  // can't set headers if response has already started
+                    _context.Response.ContentLength = null;
```

Comment:
If we can't set the ContentLength anymore, is it still safe to do the browser injection? For example, if a ContentLength was already sent, and we increase the content length by adding more content, doing that could break the code that receives the response.

Reply:
In my testing, `Response.HasStarted` was only `true` when the app with Swagger API also used the `ResponseCompression` middleware. In those cases, the agent has already been injected. Without `ResponseCompression` middleware, I didn't find any cases where `Response.HasStarted` was already true. 

I could add an additional defensive check in `WriteAsync()` so that we don't attempt to inject if the response has already started, but I don't really think it will do anything useful.

Reply:
Further testing shows that `Response.HasStarted` is always `true` when `WriteAsync()` gets called (which totally makes sense). I really think we're good at this point with the changes in this PR.